actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -539,24 +539,49 @@ def mail_to(email_address, name = nil, html_options = {}, &block)
       #   current_page?('http://www.example.com/shop/checkout?order=desc&page=1')
       #   # => true
       #
-      # Let's say we're in the <tt>http://www.example.com/products</tt> action with method POST in case of invalid product.
+      #   Different actions may share the same URL path but have a different HTTP method. Let's say we
+      #   sent a POST to <tt>http://www.example.com/products</tt> and rendered a validation error.
       #
       #   current_page?(controller: 'product', action: 'index')
       #   # => false
       #
+      #   current_page?(controller: 'product', action: 'create')
+      #   # => false
+      #
+      #   current_page?(controller: 'product', action: 'create', method: :post)
+      #   # => true
+      #
+      #   current_page?(controller: 'product', action: 'index', method: [:get, :post])
+      #   # => true
+      #
       # We can also pass in the symbol arguments instead of strings.
       #
-      def current_page?(options = nil, check_parameters: false, **options_as_kwargs)
+      def current_page?(options = nil, check_parameters: false, method: :get, **options_as_kwargs)
         unless request
           raise "You cannot use helpers that need to determine the current " \
                 "page unless your view context provides a Request object " \
                 "in a #request method"
         end
 
-        return false unless request.get? || request.head?
-
         options ||= options_as_kwargs
-        check_parameters ||= options.is_a?(Hash) && options.delete(:check_parameters)
+        if options.is_a?(Hash)
+          check_parameters ||= options.delete(:check_parameters)
+          method ||= options.delete(:method)
```

Comment:
I don't think you can do `||=` here. it works for `check_parameters` because it defaults to `false`, bit `method` defaults to a truthy value.